x-pack/filebeat/input/awss3/config.go

```diff
@@ -78,6 +79,13 @@ func (c *config) Validate() error {
 		return fmt.Errorf("number_of_workers <%v> must be greater than 0", c.NumberOfWorkers)
 	}
 
+	if c.QueueURL != "" {
+		region, _ := getRegionFromQueueURL(c.QueueURL, c.AWSConfig.Endpoint)
```

Comment:
Do we need to have this validation or could we relax it? As a user, I think the semantics of explicitly setting a region name would be my preference in my config. (It's just a shame that we cannot rely on the AWS SDK config for this because it [injects](https://github.com/elastic/beats/blob/e7e6dacfca2055d14fba269253dade2d197906d8/x-pack/libbeat/common/aws/credentials.go#L55) a default region so we can't tell if the user explicitly configured it).

This will steer users that are having trouble with their combination of queue_url and endpoint toward the region_name which is documented as for testing purposes. Not sure this is bad but I wanted to call it out. A common problem is that users get an error from getRegionFromQueueURL until they learn that they need to set `endpoint`.

Reply:
I'd be happy to relax this if we emit a log warning, though I don't think that is possible at this point, but is in input.go with some additional cost per start. WDYT?